src/diffusers/models/transformers/transformer_omnigen.py

```diff
@@ -0,0 +1,422 @@
+                offload transformer block to cpu
+            attention_kwargs: (`dict`, *optional*):
+                A kwargs dictionary that if specified is passed along to the `AttentionProcessor` as defined under
+                `self.processor` in
+                [diffusers.models.attention_processor](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/attention_processor.py).
+            return_dict (`bool`, *optional*, defaults to `True`):
+                Whether or not to return a [`OmniGen2DModelOutput`] instead of a plain tuple.
+
+        Returns:
+            If `return_dict` is True, an [`OmniGen2DModelOutput`] is returned, otherwise a `tuple` where the first
+            element is the sample tensor.
+
+        """
+
+        if attention_kwargs is not None:
+            attention_kwargs = attention_kwargs.copy()
+            lora_scale = attention_kwargs.pop("scale", 1.0)
+        else:
+            lora_scale = 1.0
+
+        if USE_PEFT_BACKEND:
+            # weight the lora layers by setting `lora_scale` for each PEFT layer
+            scale_lora_layers(self, lora_scale)
+        else:
+            if attention_kwargs is not None and attention_kwargs.get("scale", None) is not None:
+                logger.warning(
+                    "Passing `scale` via `attention_kwargs` when not using the PEFT backend is ineffective."
+                )
+        height, width = hidden_states.size()[-2:]
+        hidden_states = self.patch_embedding(hidden_states, is_input_image=False)
+        num_tokens_for_output_image = hidden_states.size(1)
+
+        time_token = self.time_token(self.time_proj(timestep).to(hidden_states.dtype)).unsqueeze(1)
+
+        condition_tokens = self.get_multimodal_embeddings(
+            input_ids=input_ids,
+            input_img_latents=input_img_latents,
+            input_image_sizes=input_image_sizes,
+        )
+        if condition_tokens is not None:
+            inputs_embeds = torch.cat([condition_tokens, time_token, hidden_states], dim=1)
+        else:
+            inputs_embeds = torch.cat([time_token, hidden_states], dim=1)
+
+        batch_size, seq_length = inputs_embeds.shape[:2]
+        position_ids = position_ids.view(-1, seq_length).long()
+
+        if attention_mask is not None and attention_mask.dim() == 3:
+            dtype = inputs_embeds.dtype
+            min_dtype = torch.finfo(dtype).min
+            attention_mask = (1 - attention_mask) * min_dtype
+            attention_mask = attention_mask.unsqueeze(1).to(inputs_embeds.dtype)
+        else:
+            raise Exception("attention_mask parameter was unavailable or invalid")
+
+        hidden_states = inputs_embeds
+
+        cos, sin = self.rotary_emb(hidden_states, position_ids)
+        for decoder_layer in self.layers:
+            hidden_states = decoder_layer(hidden_states, attention_mask=attention_mask, rotary_emb=[cos, sin])
```

Comment:
```suggestion
        if torch.is_grad_enabled() and self.gradient_checkpointing:
            hidden_states = self._gradient_checkpointing_func(decoder_layer, hidden_states, attention_mask, [cos, sin])
        else:
            hidden_states = decoder_layer(hidden_states, attention_mask=attention_mask, rotary_emb=[cos, sin])
```

To enable gradient checkpointing for training. Maybe better to first create a variable `rotary_emb = self.rotary_emb(...)` and directly use that instead